service/init.go

```diff
@@ -331,16 +377,51 @@ func (pcf *PCF) registerNF() {
 			}
 			initLog.Infof("Pcf Profile Registering to NRF: %v", profile)
 			//Indefinite attempt to register until success
-			_, self.NfId, err = consumer.SendRegisterNFInstance(self.NrfUri, self.NfId, profile)
+			profile, _, self.NfId, err = consumer.SendRegisterNFInstance(self.NrfUri, self.NfId, profile)
 			if err != nil {
 				initLog.Errorf("PCF register to NRF Error[%s]", err.Error())
 			} else {
+				if profile.HeartBeatTimer == 0 {
+					profile.HeartBeatTimer = 30
+				}
+				//sending HearBeatTimer value
+				startKeepAliveTimer <- profile.HeartBeatTimer
 				//NRF Registration Successful, Trigger for UDR Discovery
 				pcf.discoverUdr()
 			}
+		} else {
+			//stopping keepAlive timer
+			stopKeepAliveTimer <- false
+			initLog.Infof("PCF is not having Minimum Config to Register/Update to NRF")
+			problemDetails, err := consumer.SendDeregisterNFInstance()
+			if problemDetails != nil {
+				initLog.Errorf("PCF Deregister Instance to NRF failed, Problem: [+%v]", problemDetails)
+			}
+			if err != nil {
+				initLog.Errorf("PCF Deregister Instance to NRF Error[%s]", err.Error())
+			} else {
+				logger.InitLog.Infof("Deregister from NRF successfully")
+			}
 		}
 	}
+}
 
+func (pcf *PCF) updateNF() (nfProfile models.NfProfile, err error) {
+	//self := context.PCF_Self()
+	pitem := models.PatchItem{
+		Op:    "replace",
+		Path:  "/nfStatus",
+		Value: "REGISTERED",
+	}
+	var patchItem []models.PatchItem
+	patchItem = append(patchItem, pitem)
+	nfProfile, problemDetails, err := consumer.SendUpdateNFInstance(patchItem)
+	if err != nil {
+		initLog.Errorf("PCF update to NRF Error[%s]", err.Error())
```

Comment:
in case of if you are logging and in case of else you are returning error. I think we should be consistent. Either log error here or return error from here.